script/grblcon.py

```diff
@@ -0,0 +1,527 @@
+#! /usr/bin/env python3
+
+""" grblcon: an enhanced GRBL communication shell
+
+"""
+
+import threading
+from threading import Event
+import os
+import signal
+import time
+
+import serial
+import urwid
+from urwid import raw_display
+
+# TODO: Think about replacing Positions with a named tuple
+
+
+class Positions(object):
+    """ A simple object to parse and maintain positions
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self, pos_string='0,0,0,0'):
+
+        # pylint: disable=invalid-name
+        self.x, self.y, self.z, self.c = [
+            float(x) for x in pos_string.split(',')
+        ]
+
+    def __str__(self):
+        return '\n'.join([
+            'Positions',
+            '---------',
+            'X: {}'.format(self.x),
+            'Y: {}'.format(self.y),
+            'Z (Gripper): {}'.format(self.z),
+            'C (Carousel): {}'.format(self.c),
+            '\n'
+        ])
+
+
+class Status(object):
+    """Data container object which holds grbl state status and can return
+    a string appropriate for display in the status header.
+
+    """
+
+    # pylint: disable=too-few-public-methods
+
+    def __init__(self):
+        self.state = None
+        self.mach_positions = None
+        self.work_positions = None
+        self.line = None
+        self.limit_flags = None
+        self.reset()
```

Comment:
All of these variables get set properly in `.reset()`, so it's unclear to me why they need to get set here, too. Is this just to explicitly list all the member variables in the initializer?